src/diagnostics.md

```diff
@@ -315,10 +315,10 @@ reporting errors.
 
 [errors]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/index.html
 
-Diagnostics can be implemented as types which implement the `IntoDiagnostic`
+Diagnostics can be implemented as types which implement the `Diagnostic`
 trait. This is preferred for new diagnostics as it enforces a separation
 between diagnostic emitting logic and the main code paths. For less-complex
-diagnostics, the `IntoDiagnostic` trait can be derived -- see [Diagnostic
+diagnostics, the `Diagnostic` trait can be derived -- see [Diagnostic
```

Comment:
The trait is still called `IntoDiagnostic` though or am I mistaken?

Reply:
I ended up renaming the traits as `Diagnostic`, `Subdiagnostic`, and `LintDiagnostic`, to match the `derive`s.

Reply:
Ah, my bad. cc #122132. I checked https://doc.rust-lang.org/nightly/nightly-rustc 20 minutes ago when the changes weren't live yet, now they are.